addons/project/models/project.py

```diff
@@ -1366,9 +1367,43 @@ def _search_is_private(self, operator, value):
         else:
             return [('project_id', '!=', False)]
 
-    @api.depends('stage_id', 'project_id')
-    def _compute_kanban_state(self):
-        self.kanban_state = 'normal'
+    @api.depends('depend_on_ids.state', 'project_id.allow_task_dependencies')
+    def _compute_state(self):
+        for task in self:
+            if task.allow_task_dependencies:
+                for dependent_task in task.depend_on_ids:
+                    # if one of the blocking task is in a blocking state
+                    if dependent_task.state not in CLOSED_STATES:
+                        # here we check that the blocked task is not alread in a closed state (if the task is already done we don't put it in waiting state)
+                        if task.state not in CLOSED_STATES:
+                            task.state = '04_waiting_normal'
+                        return
+                # if the task as no blocking dependencies and is in waiting_normal, the task goes back to in progress
+                if task.state == '04_waiting_normal':
+                    task.state = '01_in_progress'
+
+    @api.onchange('allow_task_dependencies')
+    def _onchange_allow_task_states(self):
+        if not self.allow_task_dependencies and self.state == '04_waiting_normal':
+            self.state = '01_in_progress'
+
+    @api.onchange('stage_id')
+    def _onchange_stage_id(self):
+        if self.state != '04_waiting_normal' and self.state not in CLOSED_STATES:
+            self.state = '01_in_progress'
+
+    @api.onchange('project_id')
+    def _onchange_project_id(self):
+        if self.state != '04_waiting_normal':
+            self.state = '01_in_progress'
```

Comment:
you are right, however, I think the compute is triggered when the project changed because we have `allow_task_dependencies` (`project_id.allow_task_dependencies`) in the dependencies, so maybe this onchange is useless. 🤔